modules/core/src/lib/deck.js

```diff
@@ -287,6 +289,9 @@ export default class Deck {
 
     // Update the animation loop
     this.animationLoop.setProps(resolvedProps);
+    if ('_timeline' in props) {
+      this.animationLoop.attachTimeline(props._timeline);
```

Comment:
This is duplicating L645

Reply:
Is L645 rerun though out the lifecycle or only on initialization? The timeline can change at any time if it's supplied externally, so I'm wondering how to update the animationLoop's timeline reference in the event that happens.

Reply:
The LayerManager keeps a reference to the timeline which is used for attribute transition and ScenegraphLayer animation. The ViewManager also keeps a reference for view state transitions. If you want to be able to swap out the timeline at any time there will need to be much more code. What is the use case though?

Reply:
In my case the timeline may be nulled when a render is finished, and newly constructed when a new render job is triggered. 

Also to avoid surprises, I usually don't expect a prop needing to be treated as a singleton.

Reply:
I do see what you mean. Maybe what I'm trying to do can be approached in a different way, since I'm not concerned with view state or layer transitions. What I'm trying to do is set the time of the ScenegraphLayer animation, and interrupt/pause any ongoing transitions when a render is triggered.

Reply:
I don't think I need a timeline prop after all. I need a way to control how `this.state.animator.animate` is getting called, and understand the Scenegraph layer animation lifecycle.

https://github.com/visgl/deck.gl/blob/732a1c0ac6f0c3a47634023d69dca34b17b7d442/modules/mesh-layers/src/scenegraph-layer/scenegraph-layer.js#L247-L253

I eventually would like a way to cancel all transitions too (or document how users should approach building visualizations with both transitions and video export features). Searching for `.cancel()` shows the various managers.